pynicotine/plugins/core_commands/__init__.py

```diff
@@ -26,39 +26,637 @@ def __init__(self, *args, **kwargs):
+                "usage": ["<ip_address>"]
+            },
+            "ip": {
+                "callback": self.ip_user_command,
+                "description": _("Show IP address of user"),
+                "group": _("Network Filters"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "whois": {
+                "aliases": ["info", "w"],
+                "callback": self.whois_user_command,
+                "description": _("Show info about user"),
+                "disable": ["cli"],
+                "group": _("Users"),
+                "usage": ["<user>"],
                 "usage_private_chat": ["[user]"]
             },
-            "sample": {
-                "description": "Sample command description",
-                "aliases": ["demo"],
-                "disable": ["private_chat"],
-                "callback": self.sample_command,
-                "callback_private_chat": self.sample_command,
-                "usage": ["<choice1|choice2>", "<something..>"],
-                "usage_chatroom": ["<choice55|choice2>"]
+            "browse": {
+                "aliases": ["b"],
+                "callback": self.browse_user_command,
+                "description": _("Browse files of user"),
+                "disable": ["cli"],
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "search": {
+                "aliases": ["s"],
+                "callback": self.search_command,
+                "description": _("Start global file search"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<query>"]
+            },
+            "rsearch": {
+                "aliases": ["rs"],
+                "callback": self.search_rooms_command,
+                "description": _("Search files in joined rooms"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<query>"]
+            },
+            "bsearch": {
+                "aliases": ["bs"],
+                "callback": self.search_buddies_command,
+                "description": _("Search files of all buddies"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<query>"]
+            },
+            "usearch": {
+                "aliases": ["us"],
+                "callback": self.search_user_command,
+                "description": _("Search a user's shared files"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<\"user name\">", "<query>"]
+            },
+            "listshares": {
+                "callback": self.list_shares_command,
+                "description": _("List shares"),
+                "group": _("Shares")
+            },
+            "addshare": {
```

Comment:
share-folder (perhaps?)

Reply:
What about a single `share` command with `add` and `remove`/`rem` and `list` sub-commands?

Reply:
I'm concerned about adding too many arguments to commands, and the help message becomes confusing.

Reply:
Thats very true. Is there anything that prevents us from having commands containing spaces, such that each sub-command could then have it's own documention?